### PR TITLE
Add VTK-m implementation of radial sum

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

File: stempy/image.cpp [hunk 1]
```diff
@@ -386,11 +395,76 @@ void radialSumFrame(int centerX, int centerY, const uint16_t data[],
...
+                   radialSum);
+  }
+}
+
+#else
+void radialSumFrames(int centerX, int centerY, const uint16_t data[],
+                     int frameWidth, int frameHeight,
+                     std::vector<uint32_t>& imageNumbers,
+                     uint32_t numberOfPixels,
+                     uint32_t numberOfScanPositions RadialSum<uint64_t>& radialSum)
```
Comment (psavery) [inline stempy/image.cpp]: Haven't tried compiling these yet, but it looks like a comma is missing here.

File: stempy/image.cpp [hunk 2]
```diff
@@ -386,11 +395,76 @@ void radialSumFrame(int centerX, int centerY, const uint16_t data[],
...
+  using ControlSignature = void(FieldIn, AtomicArrayInOut);
+
+  using ExecutionSignature = void(_1, _2, WorkIndex);
+
+  template <typename AtomicInterface>
+  VTKM_EXEC void operator()(const vtkm::UInt16& value,
+                            AtomicInterface& radialSum, vtkm::Id i) const
+  {
+    auto x = i % m_shape[0];
+    auto y = i / m_shape[0];
```
Comment (psavery) [inline stempy/image.cpp]: Is `m_shape[1]` supposed to be used anywhere? I haven't seen it being used.

File: stempy/image.cpp [hunk 3]
```diff
@@ -457,6 +533,13 @@ RadialSum<uint64_t> radialSum(InputIt first, InputIt last, int scanWidth, int sc
   ThreadPool pool(numThreads);
   RadialSum<uint64_t> radialSum(scanWidth, scanHeight, maxRadius+1);
 
+#ifdef VTKm
+  // We need the reinterpret_cast as vtkm currently doesn't support atomic
+  // access for uint64.
+  auto radialSumHandle = vtkm::cont::make_ArrayHandle(
+    reinterpret_cast<vtkm::Int64*>(radialSum.data.get()),
```
Comment (psavery) [inline stempy/image.cpp]: Should this be `vtkm::UInt64*` instead?

Comment (psavery) [reply stempy/image.cpp]: Oh okay, I see your comment above now.

File: stempy/image.cpp (hunk 1 above)
Comment (psavery) [reply stempy/image.cpp]: Indeed, I tried compiling this with VTKm turned off. I get a compilation error here, and I think one or two others. Might be good to check them out.

File: stempy/image.cpp (hunk 2 above)
Comment (cjh1) [reply stempy/image.cpp]: Nope, I guess I can just pass in the width

File: stempy/image.cpp (hunk 1 above)
Comment (cjh1) [reply stempy/image.cpp]: Will do

Comment (cjh1) [reply stempy/image.cpp]: Done

File: stempy/image.cpp (hunk 2 above)
Comment (cjh1) [reply stempy/image.cpp]: Done

**Pull request merged**